src/renderer/base/renderer.cpp

```diff
@@ -78,8 +78,17 @@ Renderer::~Renderer()
             {
                 if (--tries == 0)
                 {
-                    FAIL_FAST_HR_MSG(E_UNEXPECTED, "A rendering engine required too many retries.");
+                    // Stop trying.
+                    _pThread->DisablePainting();
+                    if (_pfnRendererEnteredErrorState)
+                    {
+                        _pfnRendererEnteredErrorState();
+                    }
+                    return S_FALSE;
                 }
+                // Add a bit of backoff.
+                // Sleep 150ms, 300ms, 450ms before failing out and disabling the renderer.
+                Sleep(150 * (maxRetriesForRenderEngine - tries));

```

Comment:
Any particular reason for the backoff like this? just to give it another chance?

Reply:
Under the idea that the graphics driver probably went down for a real reason, and that restarting the card might take some time if that's what it needs to do.